src/main/java/teammates/ui/controller/InstructorFeedbackResultsPageData.java

```diff
@@ -25,94 +53,1321 @@
+            addParticipantPanelToSectionPanel(sectionPanel, currentTeam, recipientPanel);
+            teamMembersWithResponses.add(recipientIdentifier);
+            
+            prevTeam = currentTeam;
+            prevSection = currentSection;
+        }
+        
+        buildMissingParticipantPanelsForTeam(sectionPanel, prevTeam, teamMembersWithResponses, viewType.isFirstGroupedByGiver());
+        
+        teamsWithResponses.add(prevTeam);
+        buildMissingTeamAndParticipantPanelsForSection(sectionPanel, prevSection, teamsWithResponses, viewType.isFirstGroupedByGiver());
+        
+        finaliseBuildingSectionPanel(sectionPanel, prevSection, responsesGroupedByTeam, teamsWithResponses);
+        sectionPanels.put(prevSection, sectionPanel);        
+
+        // TODO introduce enums for this, because this causes problems if there is a section named "All"
+        if (selectedSection.equals("All")) {
+            sectionsWithResponses.add(prevSection); // for the last section having responses 
+            buildSectionPanelsForMissingSections(sectionsWithResponses);
+        }
+    }
+    
+    private InstructorFeedbackResultsGroupByQuestionPanel buildParticipantGroupByQuestionPanel(
+                                    String participantIdentifier,
+                                    Map.Entry<String, Map<FeedbackQuestionAttributes, List<FeedbackResponseAttributes>>> responsesForParticipant,
+                                    String additionalInfoId, int participantIndex) {
+        List<InstructorResultsQuestionTable> questionTables = new ArrayList<InstructorResultsQuestionTable>();
+        
+        int questionIndex = 0;
+        for (Map.Entry<FeedbackQuestionAttributes, List<FeedbackResponseAttributes>> responsesForParticipantForQuestion : 
+                                                                                     responsesForParticipant.getValue().entrySet()) {
+            if (responsesForParticipantForQuestion.getValue().isEmpty()) {
+                // participant has no responses for the current question
+                continue;
+            }
+            
+            questionIndex += 1;
+            
+            FeedbackQuestionAttributes currentQuestion = responsesForParticipantForQuestion.getKey();
+            List<FeedbackResponseAttributes> responsesForQuestion = responsesForParticipantForQuestion.getValue();
+
+            InstructorResultsQuestionTable questionTable 
+                = buildQuestionTableAndResponseRows(currentQuestion, responsesForQuestion,
+                                                    String.format(additionalInfoId, participantIndex, questionIndex), 
+                                                    participantIdentifier, true);
+            questionTable.setBoldQuestionNumber(false);
+            questionTables.add(questionTable);
+      
+        }
+        
+        InstructorResultsQuestionTable.sortByQuestionNumber(questionTables);
```

Comment:
Should this be done with comparators and `Collections.sort()` like in the bundle?
I'm fine either way, though.